### ISSUE TITLE
Isoseq3 tag with dual-UMI

### ISSUE THREAD

**Issue by AbeerMM** [PacificBiosciences/pbbioconda#535]
I have Pacbio subreads with dual-UMI:
5' --- 18UMI--- transcript ---18UMI--- 3'
[NNNYRNNNYRNNNYRNNN]

I tried to run isoseq3 tag using the option --design "18U-T-18U", but I keep getting this error:
`>|> 20220913 15:04:52.394 -|- FATAL -|- Run -|- 0x7fe5cdccb240|| -|- isoseq tag ERROR: Cannot set UMI twice with tags U and/or G!`

Does isoseq3 tag/dedup handle this type of dual-UMI? 

Is there any suggested workflow that works with dual-UMI starting from Pacbio subreads (bam)?

Thank you

**Comment by Magdoll** [comment]
Hi @AbeerMM ,

`isoseq3 tag` current does not support dual UMIs.

Also, please use CCS/HiFi reads only, not subreads :) if you have subreads please run CCS first!

Also this is untested, but assuming you got the CCS reads, one idea is try to run `isoseq3 tag` twice - once to trim off the first UMI, then manually manipulate the BAM file to change the UMI BAM tag to a different tag (it is the XM tag, see [FAQ](https://isoseq.how/isoseq-tags.html)), then run `isoseq3 tag` a second time. Later you can manually put back the two tags into a single XM tag if you wish so - hopefully that makes it possible to run downstream steps like `isoseq3 groupdedup`.

Let us know how it goes :)
-Liz

**Comment by AbeerMM** [comment]
> Hi @AbeerMM ,
> 
> `isoseq3 tag` current does not support dual UMIs.
> 
> Also, please use CCS/HiFi reads only, not subreads :) if you have subreads please run CCS first!
> 
> Also this is untested, but assuming you got the CCS reads, one idea is try to run `isoseq3 tag` twice - once to trim off the first UMI, then manually manipulate the BAM file to change the UMI BAM tag to a different tag (it is the XM tag, see [FAQ](https://isoseq.how/isoseq-tags.html)), then run `isoseq3 tag` a second time. Later you can manually put back the two tags into a single XM tag if you wish so - hopefully that makes it possible to run downstream steps like `isoseq3 groupdedup`.
> 
> Let us know how it goes :) -Liz

Thank you so much for your suggestion @Magdoll, I'll give it a try and see how it goes. 

Yes, I run isoseq3 tag after CCS and lima. So, my workflow is like this: CCS --> lima --> tag --> refine --> dedup. 
I noticed after dedup the transcripts have different formatting compared to isoseq3 cluster (read ID is molecule/#####, while cluster gives transcript/#####). 
I have two samples one with UMI and the other without, and I'm trying to compare them to assess the UMI. For the one without UMI, I run cluster --> collapse. For the one with UMI, I'm not sure what next comes after dedup (I need both to have collapsed clusters/transcripts to run SQANTI3 analysis). 

I installed SMRTlink10.2 (Conda) since I'm processing sequel I data, and isoseq3 (v3.5.0) is included in the installation. However, I noticed some commands are either removed or my installation has issues. For example, I can't find these command anymore:
- isoseq3 polish
- isoseq3 groupdedup
- isoseq3 correct 
Are they included in the latest version of isoseq3 (v3.8.0)? I also don't see my current version in the releases list. Not sure if I need to upgrade.
